api/secrets/secret_handler.go

```diff
@@ -0,0 +1,616 @@
+		if err != nil {
+			return nil, err
+		}
+		for _, secret := range secrets {
+			secretDTOsMap[secret.Name] = secret
+		}
+	}
+
+	for _, job := range activeDeployment.Spec.Jobs {
+		secrets, err := eh.getSecretsFromComponentObjects(&job, envNamespace)
+		if err != nil {
+			return nil, err
+		}
+		for _, secret := range secrets {
+			secretDTOsMap[secret.Name] = secret
+		}
+	}
+
+	return secretDTOsMap, nil
+}
+
+func (eh SecretHandler) getSecretsFromAuthenticationClientCertificate(activeDeployment *v1.RadixDeployment, envNamespace string) (map[string]models.Secret, error) {
+	secretDTOsMap := make(map[string]models.Secret)
+
+	for _, component := range activeDeployment.Spec.Components {
+		secret := eh.getSecretsFromComponentAuthenticationClientCertificate(&component, envNamespace)
+		if secret != nil {
+			secretDTOsMap[secret.Name] = *secret
+		}
+	}
+
+	return secretDTOsMap, nil
+}
+
+func (eh SecretHandler) getSecretRefsSecrets(radixDeployment *v1.RadixDeployment) (map[string]models.Secret, error) {
+	secretsMap := make(map[string]models.Secret)
+	for _, component := range radixDeployment.Spec.Components {
+		getRadixCommonComponentSecretRefs(&component, secretsMap)
+	}
+	for _, jobComponent := range radixDeployment.Spec.Jobs {
+		getRadixCommonComponentSecretRefs(&jobComponent, secretsMap)
+	}
+	return secretsMap, nil
+}
+
+func getRadixCommonComponentSecretRefs(component v1.RadixCommonDeployComponent, secretsMap map[string]models.Secret) {
+	secretRefs := component.GetSecretRefs()
+	for _, azureKeyVault := range secretRefs.AzureKeyVaults {
+		for _, item := range azureKeyVault.Items {
+			itemType := string(v1.RadixAzureKeyVaultObjectTypeSecret)
+			if item.Type != nil {
+				itemType = string(*item.Type)
+			}
+			secret := &models.Secret{
+				Name:        item.EnvVar,
+				DisplayName: fmt.Sprintf("%s '%s'", itemType, item.Name),
+				Type:        models.SecretTypeCsiAzureKeyVaultItem,
+				Resource:    azureKeyVault.Name,
+				Component:   component.GetName(),
+				Status:      models.External.String(),
+			}
+			secretsMap[secret.Name] = *secret
+		}
+	}
+}
+
+func (eh SecretHandler) getSecretsFromComponentAuthenticationClientCertificate(component v1.RadixCommonDeployComponent, envNamespace string) *models.Secret {
+	if auth := component.GetAuthentication(); auth != nil && component.GetPublicPort() != "" && deployment.IsSecretRequiredForClientCertificate(auth.ClientCertificate) {
+		secretName := k8sObjectUtils.GetComponentClientCertificateSecretName(component.GetName())
+		secretStatus := models.Consistent.String()
+
+		secret, err := eh.client.CoreV1().Secrets(envNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+		if err != nil {
+			secretStatus = models.Pending.String()
+		} else {
+			secretValue := strings.TrimSpace(string(secret.Data["ca.crt"]))
+			if strings.EqualFold(secretValue, secretDefaultData) {
+				secretStatus = models.Pending.String()
+			}
+		}
+
+		return &models.Secret{
```

Comment:
Seems the returned Secret object is missing many of its properties.

Reply:
It was not yet populated for some types of secrets - I just populated few. I guessed we can fulfill them on story of grouping secrets in console.